src/System.Runtime/tests/System/StringTests.cs

```diff
@@ -1607,19 +1607,31 @@ public static void Join_StringArray(string separator, string[] values, int start
         }
 
         [Fact]
-        public static void Join_StringArray_Invalid()
+        public static void Join_String_NullValues_ThrowsArgumentNullException()
         {
-            // Values is null
             Assert.Throws<ArgumentNullException>("value", () => string.Join("$$", null));
             Assert.Throws<ArgumentNullException>("value", () => string.Join("$$", null, 0, 0));
             Assert.Throws<ArgumentNullException>("values", () => string.Join("|", (IEnumerable<string>)null));
             Assert.Throws<ArgumentNullException>("values", () => string.Join<string>("|", (IEnumerable<string>)null)); // Generic overload
+        }
+
+        [Fact]
+        public static void Join_String_NegativeStartIndex_ThrowsArgumentOutOfRangeException()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => string.Join("$$", new string[] { "Foo" }, -1, 0));
+        }
 
-            Assert.Throws<ArgumentOutOfRangeException>("startIndex", () => string.Join("$$", new string[] { "Foo" }, -1, 0)); // Start index < 0
-            Assert.Throws<ArgumentOutOfRangeException>("count", () => string.Join("$$", new string[] { "Foo" }, 0, -1)); // Count < 0
+        [Fact]
+        public static void Join_String_NegativeCount_ThrowsArgumentOutOfRangeException()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("count", () => string.Join("$$", new string[] { "Foo" }, 0, -1));
+        }
```

Comment:
I've combined all the startIndex tests into a single theory test.

I kept the negaitve "count" tests on it's own, as the param name is different. Can change though if you want